homeassistant/components/mqtt/device_tracker/schema_discovery.py

```diff
@@ -63,11 +73,20 @@ async def async_setup_entry_from_discovery(hass, config_entry, async_add_entitie
         _async_setup_entity, hass, async_add_entities, config_entry=config_entry
     )
     await async_setup_entry_helper(hass, device_tracker.DOMAIN, setup, DISCOVERY_SCHEMA)
+    # (re)load legacy service
+    if MQTT_DATA_DEVICE_TRACKER_LEGACY in hass.data:
+        await async_setup_scanner_from_yaml(
+            hass, **hass.data[MQTT_DATA_DEVICE_TRACKER_LEGACY]
+        )
```

Comment:
This is a bit strange, shouldn't it happen in `mqtt/device_tracker/__init__.py` instead?

Reply:
I'll move it.